### PR TITLE
Fixed warnings

### PR DESCRIPTION
**Pull request by tivervac**
Signed-off-by: Titouan Vervack <titouan.vervack@sigasi.com>

### CODE REVIEW

File: org.eclipse.xtext/src/org/eclipse/xtext/resource/DerivedStateAwareResource.java [hunk 1]
```diff
@@ -247,7 +248,8 @@ public void installDerivedState(boolean preIndexingPhase) {
 							newFullyInitialized = false;
 							operationCanceledManager.propagateAsErrorIfCancelException(e);
 						}
-						throw Throwables.propagate(e);
+						Throwables.throwIfUnchecked(e);
```
Comment (cdietrich) [inline org.eclipse.xtext/src/org/eclipse/xtext/resource/DerivedStateAwareResource.java]: this wont work in guava 14-19

File: org.eclipse.xtext/src/org/eclipse/xtext/resource/DerivedStateAwareResource.java [hunk 2]
```diff
@@ -278,7 +280,8 @@ protected void doDiscardDerivedState() {
 			if (operationCanceledManager.isOperationCanceledException(e)) {
 				throw new IllegalStateException("IDerivedStateComputer#discardDerivedState should not check whether the current operation is canceled.", e);
 			}
-			throw Throwables.propagate(e);
+			Throwables.throwIfUnchecked(e);
```
Comment (cdietrich) [inline org.eclipse.xtext/src/org/eclipse/xtext/resource/DerivedStateAwareResource.java]: see above

File: org.eclipse.xtext/src/org/eclipse/xtext/resource/DerivedStateAwareResource.java [hunk 3]
```diff
@@ -248,8 +248,8 @@ public void installDerivedState(boolean preIndexingPhase) {
 							newFullyInitialized = false;
 							operationCanceledManager.propagateAsErrorIfCancelException(e);
 						}
-						Throwables.throwIfUnchecked(e);
-						throw new RuntimeException(e);
+						// Needed for Guava 14-19
+						Throwables.propagate(e);
```
Comment (tivervac) [inline org.eclipse.xtext/src/org/eclipse/xtext/resource/DerivedStateAwareResource.java]: @cdietrich needs `throw`

File: org.eclipse.xtext/src/org/eclipse/xtext/resource/DerivedStateAwareResource.java [hunk 4]
```diff
@@ -280,8 +280,8 @@ protected void doDiscardDerivedState() {
 			if (operationCanceledManager.isOperationCanceledException(e)) {
 				throw new IllegalStateException("IDerivedStateComputer#discardDerivedState should not check whether the current operation is canceled.", e);
 			}
-			Throwables.throwIfUnchecked(e);
-			throw new RuntimeException(e);
+			// Needed for Guava 14-19
+			Throwables.propagate(e);
```
Comment (tivervac) [inline org.eclipse.xtext/src/org/eclipse/xtext/resource/DerivedStateAwareResource.java]: @cdietrich needs `throw`

**Pull request merged**